mlflow/models/model.py

```diff
@@ -559,6 +563,20 @@ def log(
             local_path = tmp.path("model")
             run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id, metadata=metadata)
+            signature = kwargs.get("signature")
+            input_example = kwargs.get("input_example")
+            if (
+                input_example is not None
+                and signature is None
+                and hasattr(flavor, "_infer_signature")
+            ):
+                try:
+                    kwargs["signature"] = flavor._infer_signature(**kwargs)
```

Comment:
Can we call `_infer_signature` in each flavor?

Reply:
Where would each flavor call `_infer_signature`? Do you mean in `def log_model` before calling `Model.log`?

My current approach is similar to how `pyfunc.load_model` works, which calls the `def _load_pyfunc` function in each flavor.

Reply:
Can we call it in `save_model`?

Reply:
That makes sense to me. Though it would add signature inference behavior/feature to `save_model`, not just `log_model`. Does that sound good to you?